fbw-a380x/src/wasm/systems/a380_systems/src/hydraulic/autobrakes.rs

```diff
@@ -541,5 +604,777 @@ impl SimulationElement for A380AutobrakeController {
+            BTVState::RotOptimization => self.accel_during_rot_opti(),
+            _ => Acceleration::new::<meter_per_second_squared>(5.),
+        }
+    }
+
+    fn update(
+        &mut self,
+        context: &UpdateContext,
+        spoilers_active: bool,
+        braking_distance: &BrakingDistanceCalculator,
+        ground_speed: Velocity,
+        rop: &AutobrakeRunwayOverrunProtection,
+    ) {
+        self.distance_to_rwy_end = rop.distance_to_runway_end();
+
+        self.wet_prediction = braking_distance.wet_landing();
+        self.dry_prediction = braking_distance.dry_landing();
+
+        self.spoilers_active = spoilers_active;
+        self.actual_deceleration = context.long_accel();
+
+        self.integrate_distance(context, ground_speed);
+
+        self.compute_decel(ground_speed);
+
+        self.state = self.update_state(ground_speed);
+    }
+
+    fn braking_distance_remaining(&self) -> Length {
+        let distance_remaining_raw = if self.is_oans_fallback_mode() {
+            self.in_flight_btv_stopping_distance.value() - self.rolling_distance
+        } else {
+            self.oans_distance_to_exit.value()
+        };
+
+        let distance_from_btv_exit = Length::new::<meter>(Self::DISTANCE_OFFSET_TO_RELEASE_BTV_M);
+
+        // Max distance clamped to end of rwy minus a margin
+        (distance_remaining_raw - distance_from_btv_exit)
+            .max(Length::default())
+            .min(
+                self.distance_to_rwy_end
+                    - Length::new::<meter>(
+                        Self::REMAINING_BRAKING_DISTANCE_END_OF_RUNWAY_OFFSET_METERS,
+                    ),
+            )
+    }
+
+    fn compute_decel(&mut self, ground_speed: Velocity) {
+        match self.state {
+            BTVState::RotOptimization
+            | BTVState::Decel
+            | BTVState::EndOfBraking
+            | BTVState::_OutOfDecelRange => {
+                let speed_at_btv_release =
+                    Velocity::new::<meter_per_second>(Self::TARGET_SPEED_TO_RELEASE_BTV_M_S)
+                        * Self::SAFETY_RATIO_ON_RELEASE_SPEED;
+
+                self.final_distance_remaining = self.braking_distance_remaining();
+
+                let delta_speed_to_achieve = ground_speed - speed_at_btv_release;
+
+                let target_deceleration_raw =
+                    -delta_speed_to_achieve.get::<meter_per_second>().powi(2)
+                        / (2. * self.final_distance_remaining.get::<meter>());
+
+                let target_deceleration_safety_corrected =
+                    target_deceleration_raw * self.safety_margin();
+
+                self.deceleration_request = Acceleration::new::<meter_per_second_squared>(
+                    target_deceleration_safety_corrected
+                        .max(self.desired_deceleration.get::<meter_per_second_squared>())
+                        .min(5.),
+                );
+            }
+            _ => {
```

Comment:
This is against the coding guidelines: https://github.com/flybywiresim/aircraft/blob/master/fbw-common/src/wasm/systems/guidelines.md#explicit-match-pattern-declaration